src/store/getters/state-getters.ts

```diff
@@ -440,8 +456,21 @@ export const hasContactInfoChanged = (state: StateIF): boolean => {
 
 /** Whether share structure data has changed. */
 export const hasShareStructureChanged = (state: StateIF): boolean => {
-  const currentShareClasses = { ...omit(state.stateModel.shareStructureStep.shareClasses, ['action']) }
-  const originalShareClasses = { ...omit(state.stateModel.originalSnapshot?.shareStructure.shareClasses, ['action']) }
+  let currentShareClasses = getShareClasses(state)
+  let originalShareClasses = state.stateModel.originalSnapshot?.shareStructure.shareClasses
+
+  // Null action properties can be assigned to the ShareClasses when cancelling edits
+  // This is fail safe to ensure null actions are not included in the comparison
```

Comment:
Why not just use `omit()` as before?

Reply:
(I'm asking, not telling.)

Reply:
PS Very helpful code comment - thanks!

Reply:
My previous code wasn't working :( It was actually omitting `action` on the array level and not within the individual objects. 
Also, we don't actually want to omit all the actions because a `removed` for instance, we'd still want to see until the point of filing for the purposes of display/restore etc..

It's been quite complicated this part of the code 😅

Reply:
Yes, it looks complicated, and I wish it didn't.